### ISSUE TITLE
[ui5-calendar-legend] missing accessibility semantics

### ISSUE THREAD

**Issue by unazko** [SAP/ui5-webcomponents#10298]
### Bug Description

The `ui5-calendar-legend` is missing a role description of "Calendar Legend" and the following accessibility semantics:
- The component's root container should have a role "List"
- The calendar items are represented as list items and should have a role "listitem"

For comparison checkout the sap.ui.unified.CalendarLegend control:
https://ui5.sap.com/#/entity/sap.ui.unified.Calendar/sample/sap.ui.unified.sample.CalendarSpecialDaysLegend

### Affected Component

ui5-calendar-legend

### Expected Behaviour

The `ui5-calendar-legend` as an interactive component should have a correct accessibility semantics.

### Isolated Example

https://sap.github.io/ui5-webcomponents/components/CalendarLegend/

### Steps to Reproduce

Open the following sample https://sap.github.io/ui5-webcomponents/components/CalendarLegend/
and observe the calendar legend items in the browser debugger tools in the accessibility tab.

Compare between UI5:
![UI5AccSemantics](https://github.com/user-attachments/assets/bd3f772f-ce7b-49b8-b26d-82475b9d48f9)
and ui5 webc:
![WebcAccSemantics](https://github.com/user-attachments/assets/76871b24-e925-4ef3-b7bb-19a453b067c6)

### Priority

Medium

### UI5 Web Components Version

latest

### Browser

Chrome

### Operating System

Windows

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by unazko** [comment]
Hello @SAP/ui5-webcomponents-topic-b,

This issue is for us to fix.

Best regards,
Boyan
